python-sdk/nuimages/nuimages.py

```diff
@@ -701,52 +794,58 @@ def render_image(self,
-                name = annotation_name(attributes, category_name, with_attributes=with_attributes)
-                if ann['mask'] is None:
-                    continue
-                mask = mask_decode(ann['mask'])
-
-                # Draw mask, rectangle and text.
-                draw.bitmap((0, 0), Image.fromarray(mask * 128), fill=tuple(color + (128,)))
-                draw.rectangle(bbox, outline=color, width=box_line_width)
-                if with_category:
-                    draw.text((bbox[0], bbox[1]), name, font=font)
+        with_annotations_options = ['all', 'surfaces', 'objects', 'none']
+        assert with_annotations in with_annotations_options, \
+            'Error: {} is not a valid option for with_annotations. ' \
+            'Only {} are allowed.'.format(with_annotations, with_annotations_options)
+        if with_annotations is not 'none':
+            if with_annotations == 'all' or with_annotations == 'surfaces':
+                # Load stuff / surface regions.
+                surface_anns = [o for o in self.surface_ann if o['sample_data_token'] == sd_token_camera]
+                if surface_tokens is not None:
+                    surface_anns = [o for o in surface_anns if o['token'] in surface_tokens]
+
+                # Draw stuff / surface regions.
+                for ann in surface_anns:
+                    # Get color and mask.
+                    category_token = ann['category_token']
+                    category_name = self.get('category', category_token)['name']
+                    color = self.color_map[category_name]
+                    if ann['mask'] is None:
+                        continue
+                    mask = mask_decode(ann['mask'])
+
+                    # Draw mask. The label is obvious from the color.
+                    draw.bitmap((0, 0), Image.fromarray(mask * 128), fill=tuple(color + (128,)))
+
+            if with_annotations == 'all' or with_annotations == 'objects':
+                # Load object instances.
+                object_anns = [o for o in self.object_ann if o['sample_data_token'] == sd_token_camera]
+                if object_tokens is not None:
+                    object_anns = [o for o in object_anns if o['token'] in object_tokens]
+
+                # Draw object instances.
+                for ann in object_anns:
+                    # Get color, box, mask and name.
+                    category_token = ann['category_token']
+                    category_name = self.get('category', category_token)['name']
+                    color = self.color_map[category_name]
+                    bbox = ann['bbox']
+                    attr_tokens = ann['attribute_tokens']
+                    attributes = [self.get('attribute', at) for at in attr_tokens]
+                    name = annotation_name(attributes, category_name, with_attributes=with_attributes)
+                    if ann['mask'] is None:
```

Comment:
I personally will avoid using continue in a loop if it is possible, I think if ann['mask'] is not None is more readable here.